sdk/storage/Azure.Storage.Files/src/Models/FileSmbProperties.cs

```diff
@@ -0,0 +1,166 @@
+        }
+
+        /// <summary>
+        /// Checks if two FileSmbProperties are equal.
+        /// </summary>
+        /// <param name="other"></param>
+        /// <returns></returns>
+        public override bool Equals(object other)
+            => other is FileSmbProperties && this.Equals((FileSmbProperties)other);
+
+        /// <summary>
+        /// Gets the hash code for the FileSmbProperties.
+        /// </summary>
+        /// <returns></returns>
+        public override int GetHashCode()
+            => this.FileAttributes.GetHashCode()
+            ^ this.FilePermissionKey.GetHashCode()
+            ^ this.FileCreationTime.GetHashCode()
+            ^ this.FileLastWriteTime.GetHashCode()
+            ^ this.FileChangeTime.GetHashCode()
+            ^ this.FileId.GetHashCode()
+            ^ this.ParentId.GetHashCode();
+
+        /// <summary>
+        /// Check if two FileSmbProperties instances are equal.
+        /// </summary>
+        /// <param name="left">The first instance to compare.</param>
+        /// <param name="right">The second instance to compare.</param>
+        /// <returns>True if they're equal, false otherwise.</returns>
+        public static bool operator ==(FileSmbProperties left, FileSmbProperties right) => left.Equals(right);
+
+        /// <summary>
+        /// Check if two FileSmbProperties instances are not equal.
+        /// </summary>
+        /// <param name="left">The first instance to compare.</param>
+        /// <param name="right">The second instance to compare.</param>
+        /// <returns>True if they're not equal, false otherwise.</returns>
+        public static bool operator !=(FileSmbProperties left, FileSmbProperties right) => !(left == right);
+
+        /// <summary>
+        /// Check if two FileSmbProperties instances are equal.
+        /// </summary>
+        /// <param name="other">The instance to compare to.</param>
+        public bool Equals(FileSmbProperties other)
+            => this.FileAttributes == other.FileAttributes
+            && this.FilePermissionKey == other.FilePermissionKey
+            && this.FileCreationTime == other.FileCreationTime
+            && this.FileLastWriteTime == other.FileLastWriteTime
+            && this.FileChangeTime == other.FileChangeTime
+            && this.FileId == other.FileId
+            && this.ParentId == other.ParentId;
+
+        internal string FileCreationTimeToString()
+            => this.FileCreationTime.HasValue ? DateTimeOffSetToString(this.FileCreationTime.Value) : null;
+
+        internal string FileLastWriteTimeToString()
+            => this.FileLastWriteTime.HasValue ? DateTimeOffSetToString(this.FileLastWriteTime.Value) : null;
+
+        private static string DateTimeOffSetToString(DateTimeOffset dateTimeOffset)
+            => dateTimeOffset.UtcDateTime.ToString("yyyy'-'MM'-'dd'T'HH':'mm':'ss'.'fffffff'Z'", CultureInfo.InvariantCulture);
```

Comment:
Either move this value into `Constants` or reuse an existing 8601 constant.  And just to clarify, the service will accept `Now`/`Preserve` but never return them?

Reply:
The service will never return Now or Preserve.  Moved to Constants.